### PR TITLE
Add new keyboard "Ergomirage"

### PR DESCRIPTION
**Pull request by takashicompany**
Hi.
I designed new keyboard "Ergomirage".
Please review this pr.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/takashicompany/ergomirage/config.h
Comment (drashna) [inline keyboards/takashicompany/ergomirage/config.h:1]: This file could be removed.

File: keyboards/takashicompany/ergomirage/info.json [hunk 1]
```diff
@@ -0,0 +1,400 @@
+{
+    "manufacturer": "takashicompany",
+    "keyboard_name": "Ergomirage",
+    "maintainer": "takashicompany",
+    "bootloader": "caterina",
```
Comment (drashna) [inline keyboards/takashicompany/ergomirage/info.json:5]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/takashicompany/ergomirage/info.json [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+            "F5",
+            "F6",
+            "F7",
+            "B1",
+            "B3",
+            "B2",
+            "B6"
+        ]
+    },
+    "processor": "atmega32u4",
```
Comment (drashna) [inline keyboards/takashicompany/ergomirage/info.json:35]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/ergomirage/config.h
Comment (drashna) [inline keyboards/takashicompany/ergomirage/config.h:1]: This file can be removed.  It's not doing anything, and is extra at this point.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/ergomirage/config.h
Comment (takashicompany) [reply keyboards/takashicompany/ergomirage/config.h:1]: I Deleted.

**Review by drashna — APPROVED** [review]

File: keyboards/takashicompany/ergomirage/info.json [hunk 3]
```diff
@@ -0,0 +1,399 @@
+{
```
Comment (fauxpark) [inline keyboards/takashicompany/ergomirage/info.json:1]: Please run `qmk format-json`.

File: keyboards/takashicompany/ergomirage/info.json [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+                    "x": 2,
+                    "y": 1.25
+                },
+                {
+                    "matrix": [
+                        1,
+                        0
+                    ],
+                    "x": 0,
+                    "y": -10.5
```
Comment (fauxpark) [inline keyboards/takashicompany/ergomirage/info.json:66]: `x` and `y` positions cannot be negative. Your layout looks like this:
image

Comment (takashicompany) [reply keyboards/takashicompany/ergomirage/info.json]: I fixed https://github.com/qmk/qmk_firmware/pull/20655/commits/5116eb7aa55a6b7640829baf726e67a0e469f532

File: keyboards/takashicompany/ergomirage/info.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                { "matrix": [5, 2], "x": 12, "y": 3.25 },
+                { "matrix": [5, 5], "x": 13, "y": 3.25 },
+                { "matrix": [6, 0], "x": 2.5, "y": 4 },
+                { "matrix": [6, 1], "x": 3.5, "y": 4 },
+                { "matrix": [6, 2], "x": 4.5, "y": 4 },
+                { "matrix": [6, 3], "x": 5.5, "y": 4 },
+                { "matrix": [7, 0], "x": 7.5, "y": 4 },
+                { "matrix": [7, 1], "x": 8.5, "y": 4 },
+                { "matrix": [7, 2], "x": 9.5, "y": 4 },
+                { "matrix": [7, 3], "x": 10.5, "y": 4 }
```
Comment (fauxpark) [inline keyboards/takashicompany/ergomirage/info.json:71]:
```suggestion
                {"matrix": [0, 0], "x": 0, "y": 1.25},
                {"matrix": [0, 3], "x": 1, "y": 1.25},
                {"matrix": [1, 0], "x": 2, "y": 0.25},
                {"matrix": [1, 3], "x": 3, "y": 0},
                {"matrix": [2, 0], "x": 4, "y": 0.25},
                {"matrix": [2, 3], "x": 5, "y": 0.5},

                {"matrix": [3, 0], "x": 8, "y": 0.5},
                {"matrix": [3, 3], "x": 9, "y": 0.25},
                {"matrix": [4, 0], "x": 10, "y": 0},
                {"matrix": [4, 3], "x": 11, "y": 0.25},
                {"matrix": [5, 0], "x": 12, "y": 1.25},
                {"matrix": [5, 3], "x": 13, "y": 1.25},

                {"matrix": [0, 1], "x": 0, "y": 2.25},
                {"matrix": [0, 4], "x": 1, "y": 2.25},
                {"matrix": [1, 1], "x": 2, "y": 1.25},
                {"matrix": [1, 4], "x": 3, "y": 1},
                {"matrix": [2, 1], "x": 4, "y": 1.25},
                {"matrix": [2, 4], "x": 5, "y": 1.5},

                {"matrix": [3, 1], "x": 8, "y": 1.5},
                {"matrix": [3, 4], "x": 9, "y": 1.25},
                {"matrix": [4, 1], "x": 10, "y": 1},
                {"matrix": [4, 4], "x": 11, "y": 1.25},
                {"matrix": [5, 1], "x": 12, "y": 2.25},
                {"matrix": [5, 4], "x": 13, "y": 2.25},

                {"matrix": [0, 2], "x": 0, "y": 3.25},
                {"matrix": [0, 5], "x": 1, "y": 3.25},
                {"matrix": [1, 2], "x": 2, "y": 2.25},
                {"matrix": [1, 5], "x": 3, "y": 2},
                {"matrix": [2, 2], "x": 4, "y": 2.25},
                {"matrix": [2, 5], "x": 5, "y": 2.5},

                {"matrix": [3, 2], "x": 8, "y": 2.5},
                {"matrix": [3, 5], "x": 9, "y": 2.25},
                {"matrix": [4, 2], "x": 10, "y": 2},
                {"matrix": [4, 5], "x": 11, "y": 2.25},
                {"matrix": [5, 2], "x": 12, "y": 3.25},
                {"matrix": [5, 5], "x": 13, "y": 3.25},

                {"matrix": [6, 0], "x": 2.5, "y": 4},
                {"matrix": [6, 1], "x": 3.5, "y": 4},
                {"matrix": [6, 2], "x": 4.5, "y": 4},
                {"matrix": [6, 3], "x": 5.5, "y": 4},

                {"matrix": [7, 0], "x": 7.5, "y": 4},
                {"matrix": [7, 1], "x": 8.5, "y": 4},
                {"matrix": [7, 2], "x": 9.5, "y": 4},
                {"matrix": [7, 3], "x": 10.5, "y": 4}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**